src/components/FindMovie/FindMovie.tsx

```diff
@@ -1,10 +1,70 @@
-import React from 'react';
+import React, { ChangeEvent, useState, FormEvent } from 'react';
+import classNames from 'classnames';
+import { getMovie } from '../../api';
+import { MovieCard } from '../MovieCard';
 import './FindMovie.scss';
+import { Movie } from '../../types/Movie';
+
+type Props = {
+  addMovie: (movie: Movie) => void,
+};
+
+export const FindMovie: React.FC<Props> = ({ addMovie }) => {
+  const [query, setQuery] = useState('');
+  const [movie, setMovie] = useState<Movie | null>(null);
+  const [loader, setLoader] = useState(false);
+  const [error, setError] = useState(false);
+
+  const handleAddMovie = () => {
+    if (movie) {
+      addMovie(movie);
+    }
+
+    setQuery('');
+    setMovie(null);
+  };
+
+  const handleChangeValue = (event: ChangeEvent<HTMLInputElement>) => {
+    setQuery(event.currentTarget.value);
+    setError(false);
+  };
+
+  const handleSubmit = (event: FormEvent) => {
+    event.preventDefault();
+    setLoader(true);
+
+    getMovie(query)
+      .then(response => {
```

Comment:
Let’s use async await. It gives you a simpler way to work with asynchronous promise-based code